draft-ietf-jsonpath-base.md

```diff
@@ -1568,11 +1588,31 @@ If the nodelist contains one or more nodes, the result is
 `LogicalTrue`.
 If the nodelist is empty, the result is `LogicalFalse`.
 
+### `single` Function Extension {#single-fn}
+
+Parameters:
+: 1. `NodesType`
+
+Result:
+: `SingleNodeType`
+
+The "single" function extension provides a way to obtain a
+node in a nodelist and make that available for further
+processing in the filter expression:
+
+~~~ JSONPath
+$[?value(single(fn(@.*.price))) >= 7.99]
+~~~
+
+Its only argument is a nodelist.
+If the nodelist contains a single node, the result is the value of the node.
```

Comment:
That's what "value" is supposed to do.
"single" is useful for finding out whether a nodelist is singular

Use `single($.a.*.author)` to see whether there is only one author

Reply:
The point of this PR is to explore what it looks like to mirror (on the function side) the fact that expressions like `$[?@.*.price >= 7.99]` are not well-formed and to avoid what, in my opinion, is a surprise in #410 of `value` converting a multi-node nodelist to `Nothing`.

Feel free to add a function like `single` (presumably returning `LogicalType`) in #410, but in this PR, that's not the purpose of `single()`, as defined.